lib/checks/checks.go

```diff
@@ -59,34 +59,52 @@ func New(config Config) (*checker, error) {
 	return &checker{Config: config}, nil
 }
 
-// ValidateManifest verifies the specified manifest against the host environment.
+// Validate verifies the specified manifest against the host environment.
 // Returns list of failed health probes.
-func ValidateManifest(
-	manifest schema.Manifest,
-	profile schema.NodeProfile,
-	dockerConfig storage.DockerConfig,
-	stateDir string,
-) (failedProbes []*agentpb.Probe, err error) {
+func (r *ManifestValidator) Validate(ctx context.Context) (failedProbes []*agentpb.Probe, err error) {
 	var errors []error
-	failed, err := schema.ValidateRequirements(profile.Requirements, stateDir)
+	requirements := r.Profile.Requirements
+	if len(r.Mounts) != 0 {
+		requirements = overrideMounts(requirements, r.Mounts)
+	}
+	failed, err := schema.ValidateRequirements(ctx, requirements, r.StateDir)
 	if err != nil {
 		errors = append(errors, trace.Wrap(err,
-			"error validating profile requirements, see syslog for details"))
+			"error validating profile requirements, see log file for details"))
```

Comment:
Nit: If this error is user-facing, it might be confusing, I would say which log file at least (/var/log/gravity-system.log?)